4.analyze_data/notebooks/linear_model/scripts/0.linear_model_concat_coeff.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Perform linear model per CellProfiler feature on concatenated normalized data from plates 5, 3 prime, and 3
+# 
+# We will include 3 co-variates:
+# 
+# 1. Cell count per well contribution
+# 2. Plate contribution
+# 3. Genotype contribution (WT versus Null)
+
+# In[1]:
+
+
+import pathlib
+
+import pandas as pd
+from pycytominer import feature_select
+from pycytominer.cyto_utils import infer_cp_features
+from sklearn.linear_model import LinearRegression
+
+
+# In[2]:
+
+
+# Define inputs and outputs
+data_dir = pathlib.Path("../../../../nf1_cellpainting_data/3.processing_features/data/single_cell_profiles/")
+cp_files = [
+    pathlib.Path(data_dir, f"Plate_{plate}_sc_normalized.parquet")
+    for plate in ["5", "3", "3_prime"]
+]
+
+# Output directory for LM coeffs
+output_dir = pathlib.Path("./results")
+output_dir.mkdir(exist_ok=True)
+
+# Name of output file with LM coeffs
+output_cp_file = pathlib.Path(
+    output_dir, "linear_model_cp_features_concat_plate5_plate3_plate3prime.tsv"
+)
+
+# Filter and load the specified files
+metadata_mapping = {"Plate_3_prime": "Plate_3_prime"}
+
+# Filter and load the specified files
+df_list = []
+
+for plate in cp_files:
+    cp_file = pathlib.Path(plate)
+
+    if cp_file.exists():
+        # Load data
+        df = pd.read_parquet(cp_file)
+
+        # Update Metadata_Plate only for Plate_3_prime
+        if plate.stem.replace("_sc_normalized", "") == "Plate_3_prime":
+            df["Metadata_Plate"] = "Plate_3_prime"
+
+        # Remove rows with 'HET' in 'Metadata_genotype'
+        df = df[df["Metadata_genotype"] != "HET"]
+
+        df_list.append(df)
+
+# Concatenate the DataFrames
+concat_df = pd.concat(df_list, ignore_index=True)
+
+# Make sure there are no NaNs
+concat_df = feature_select(concat_df, operation="drop_na_columns", na_cutoff=0)
+
+# Define CellProfiler features
+cp_features = infer_cp_features(concat_df)
+
+print(f"We are testing {len(cp_features)} CellProfiler features")
+print(concat_df.shape)
+concat_df.head()
+
+
+# ## Set up dummy framework for WT versus Null and plate to plate comparison
+
+# In[3]:
+
+
+# Setup linear modeling framework
+variables = ["Metadata_number_of_singlecells"]
+X = concat_df.loc[:, variables]
+
+# Add dummy matrix of categorical genotypes
+genotype_x = pd.get_dummies(data=concat_df.Metadata_genotype)
+
+# Add dummy matrix of categorical plates
+plate_x = pd.get_dummies(data=concat_df.Metadata_Plate)
+
+X = pd.concat([X, genotype_x, plate_x], axis=1)
+
+print(X.shape)
+X.head()
+
+
+# In[5]:
```

Comment:
This cell looked like there were a lot of things happening. If it's helpful, consider breaking things up into multiple cells.

Reply:
I could likely split this into two cells, I will consider this!